utils.go

```diff
@@ -51,6 +54,19 @@ func safeAsyncClose(b *Broker) {
 	})
 }
 
+func asyncCloseWithAck(b *Broker, wg *sync.WaitGroup) {
+	tmp := b // local var prevents clobbering in goroutine
+	wg.Add(1)
+	go func () {
+		withRecover(func() {
+			if err := tmp.Close(); err != nil {
```

Comment:
Are there cases where this doesn't terminate? Is is possible we need to either timeout on the Wait()? Also, is it possible if a Close() returns an err it will become invalid and never return success?